src/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/NetCred.cs

```diff
@@ -67,15 +67,15 @@ private void SplitUsername(string username, ref string parsedUserName, ref strin
 
             // Logon user expects the username in UPN or to have the username and domain split to the separate parameters.
             // It does not work properly with NT4 style name formats.  This function will put the username in the proper format.
-            StringBuilder splitUsername = new StringBuilder(UnsafeNativeMethods.CRED_MAX_USERNAME_LENGTH);
-            StringBuilder splitDomain = new StringBuilder(UnsafeNativeMethods.CRED_MAX_DOMAIN_TARGET_LENGTH);
+            char* splitUsername = stackalloc char[UnsafeNativeMethods.CRED_MAX_USERNAME_LENGTH];
+            char* splitDomain = stackalloc char[UnsafeNativeMethods.CRED_MAX_DOMAIN_TARGET_LENGTH];
 
```

Comment:
these allocation is more than 2K on the stack I think. I am not sure if this will be a concern if we get into deep stack.